create-api-key.md updated

jesusgonzalez12 opened the pull request:
This version introduces improvements on how a user can work with creating API keys using the pyc8 SDK.
Only changes to the pyc8 code have been introduced.
You can check for more details on the Kanban board.

jesusgonzalez12 committed:
create-api-key.md updated

This version introduces improvements on how a user can work with creating API keys using the pyc8 SDK.
Only changes to the pyc8 code have been introduced.
You can check for more details on the Kanban board.

jesusgonzalez12 commented on docs/sdks/create-api-key.md:
```diff
@@ -29,8 +29,23 @@ client = C8Client(protocol='https', host='gdn.paas.macrometa.io', port=443,
+# Checks if the API key already exists and acts accordingly
+if API_ACTIVE:
+    # Send message that the key already exists
+    print("Remove API Key: ", client.remove_api_key(CHOSEN_API_KEY_ID))
```
```suggestion
    print("Can't create an already existing key")
```

jesusgonzalez12 committed:
Update docs/sdks/create-api-key.md

jamesm-macrometa commented on docs/sdks/create-api-key.md:
```diff
@@ -29,8 +29,23 @@ client = C8Client(protocol='https', host='gdn.paas.macrometa.io', port=443,
 
-# Create an API key
-print("Create API Key: ", client.create_api_key('id1'))
+# Check wether the chosen_api_key exists or not, comparing it to the list elements
```
```suggestion
# Checks if `chosen_api_key` exists
```

jamesm-macrometa replied:
Same comment as on other PR, not sure what we mean by "list elements" but we might want to add that part back in.

jesusgonzalez12 replied:
Sorry, I missed adding a few lines of code. Fixed.

A list of all API key IDs a user has is saved in a variable which is used to make a comparison check later.

jamesm-macrometa replied:
Got it. I'm not certain it's necessary to mention that much detail in this particular message.  @diana-macrometa any thoughts?

jesusgonzalez12 replied:
I have updated the comment to improve semmantics.

jamesm-macrometa commented on docs/sdks/create-api-key.md:
```diff
@@ -29,8 +29,23 @@ client = C8Client(protocol='https', host='gdn.paas.macrometa.io', port=443,
+        API_ACTIVE = True
+        break
+
+# Checks if the API key already exists and acts accordingly
```
```suggestion
# Checks if the API key exists
```

jamesm-macrometa commented on docs/sdks/create-api-key.md:
```diff
@@ -29,8 +29,23 @@ client = C8Client(protocol='https', host='gdn.paas.macrometa.io', port=443,
+
+# Checks if the API key already exists and acts accordingly
+if API_ACTIVE:
+    # Send message that the key already exists
```
```suggestion
    # Display a message that the key already exists
```

jamesm-macrometa reviewed (changes requested):
A few minor changes, looks good!

jesusgonzalez12 committed:
Apply suggestions from code review

Co-authored-by: James <86624594+jamesm-macrometa@users.noreply.github.com>

jesusgonzalez12 committed:
Adding missing code for elements list

Adding code that saves all API key ID's a user has on a variable for comparison

jesusgonzalez12 commented:
Thanks for the review.
Please, check again the element list topic.

jesusgonzalez12 committed:
Updated semmantics on a comment

Uodated a few words of a commented line to improve semmantics.

koshy-thomas committed:
Merge branch 'master' into patch-3

jamesm-macrometa commented on docs/sdks/create-api-key.md:
```diff
@@ -29,8 +29,28 @@ client = C8Client(protocol='https', host='gdn.paas.macrometa.io', port=443,
+for api in client.list_all_api_keys():
+    apiKeyIds.append(api["keyid"])
+
+# Check wether the chosen_api_key exists or not, comparing it to the API key list
```
```suggestion
# Checks if `chosen_api_key` exists
```

jamesm-macrometa committed:
Update docs/sdks/create-api-key.md

jamesm-macrometa committed:
Merge branch 'master' into patch-3
